world_name: robotics_testlabs

# -----------------------------------------------------------------------------------------------------------------------------

plugins:
- name: sync_server_plugin
  lib: libed_sync_server_plugin.so
  frequency: 10
#- name: robot_plugin
#  lib: libed_robot_plugin.so
#  parameters:
#      robot_name: amigo
#      urdf_rosparam: /amigo/robot_description
#      joint_topics:
#        - topic: /amigo/joint_states
- name: gui_server
  lib: libed_gui_server_plugin.so
  parameters:
      robot_name: amigo
- name: navigation
  lib: libed_navigation_plugin.so
  parameters:
      occupancy_grid_publisher:
          resolution: 0.05
          frame_id: /map
          min_z: 0.05
          max_z: 1.8
- name: tf_publisher
  lib: libed_tf_publisher_plugin.so
  parameters:
      root_frame_id: /map
      exclude: amigo
- name: kinect_integration
  lib: libed_kinect_plugin.so
  parameters:
      topic: /amigo/top_kinect/rgbd
      voxel_size: 0.02
      max_range: 2.0
      clearing_padding_fraction: 0.05
      normal_k_search: 15
      visualize: 0
      association_modules:
        - type: PointNormal
          parameters:
              association_correspondence_distance: 0.3
              position_weight: 2.0
              normal_weight: 1.0
              render_width: 320
              render_max_range: 2.2
              render_voxel_size: 0.02
              normal_k_search: 15
              visualize: 0
        - type: PolygonHeight
          parameters:
              max_range: 2.0
              tolerance: 0.035
              min_cluster_size: 20
              visualize: 0
      segmentation_modules:
      - type: EuclideanClustering
        parameters:
            tolerance: 0.035
            min_cluster_size: 20
            visualize: 0
- name: localization
  lib: libed_localization_plugin.so
  parameters:
      initial_pose_topic: /amigo/initialpose
      num_particles: 500
      initial_pose:
          x: $(env ROBOT_INIT_X)
          y: $(env ROBOT_INIT_Y)
          rz: $(env ROBOT_INIT_PHI)    
      laser_model:
          topic: /amigo/base_laser/scan
          num_beams: 100
          z_hit: 0.95
          sigma_hit: 0.2
          z_short: 0.1
          z_max: 0.05
          z_rand: 0.05
          lambda_short: 0.1
          range_max: 10
          min_particle_distance: 0.01
          min_particle_rotation_distance: 0.02
      odom_model:
          map_frame: map
          odom_frame: /amigo/odom
          base_link_frame: /amigo/base_link
          alpha1: 0.05   # rot -> trans + strafe
          alpha2: 0.5    # trans -> rot
          alpha3: 0.5    # trans -> trans
          alpha4: 0.5    # rot -> rot
          alpha5: 0.5    # trans -> strafe
